cmd/bnscli/cmd_gov.go

```diff
@@ -1,112 +1,84 @@
+		descFl  = fl.String("description", "Transfer funds to distribution account", "The proposal description.")
+		eRuleFl = fl.Uint64("electionrule", 0, "The ID of the election rule to be used.")
 	)
 	fl.Parse(args)
 
-	option := app.ProposalOptions{
-		Option: &app.ProposalOptions_SendMsg{
-			SendMsg: &cash.SendMsg{
-				Metadata: &weave.Metadata{Schema: 1},
-				Src:      *srcFl,
-				Dest:     *dstFl,
-				Amount:   amountFl,
-				Memo:     *memoFl,
-				Ref:      nil,
-			},
-		},
-	}
-	rawOption, err := option.Marshal()
+	raw, err := ioutil.ReadAll(input)
 	if err != nil {
-		return fmt.Errorf("cannot serialize %T option: %s", option, err)
-	}
-	tx := &app.Tx{
-		Sum: &app.Tx_CreateProposalMsg{
-			CreateProposalMsg: &gov.CreateProposalMsg{
-				Metadata: &weave.Metadata{Schema: 1},
-				Base: &gov.CreateProposalMsgBase{
-					Title:          *titleFl,
-					Description:    *descFl,
-					StartTime:      weave.AsUnixTime(time.Now().Add(time.Minute)),
-					ElectionRuleID: sequenceID(*eRuleFl),
-				},
-				RawOption: rawOption,
-			},
-		},
+		return fmt.Errorf("cannot read transaction: %s", err)
 	}
-	raw, err := tx.Marshal()
-	if err != nil {
-		return fmt.Errorf("cannot serialize transaction: %s", err)
+	if len(raw) == 0 {
+		return errors.New("no input data")
 	}
-	_, err = output.Write(raw)
-	return err
-}
 
-func cmdNewEscrowProposal(input io.Reader, output io.Writer, args []string) error {
-	fl := flag.NewFlagSet("", flag.ExitOnError)
-	fl.Usage = func() {
-		fmt.Fprintln(flag.CommandLine.Output(), `
-Create a proposal transaction for releasing funds from given escrow.
-		`)
-		fl.PrintDefaults()
+	var tx app.Tx
+	if err := tx.Unmarshal(raw); err != nil {
+		return fmt.Errorf("cannot deserialize transaction: %s", err)
 	}
-	var (
-		escrowFl = flHex(fl, "escrow", "", "A hex encoded ID of an escrow that is to be released.")
-		amountFl = flCoin(fl, "amount", "", "Optional amount that is to be transferred from the escrow. The whole escrow hold amount is used if no value is provided.")
-		titleFl  = fl.String("title", "Transfer funds to distribution account", "The proposal title.")
-		descFl   = fl.String("description", "Transfer funds to distribution account", "The proposal description.")
-		eRuleFl  = fl.Uint64("electionrule", 0, "The ID of the election rule to be used.")
-	)
-	fl.Parse(args)
 
-	var amount []*coin.Coin
-	if !coin.IsEmpty(amountFl) {
-		amount = append(amount, amountFl)
+	msg, err := tx.GetMsg()
+	if err != nil {
+		return fmt.Errorf("cannot extract message from the transaction: %s", err)
 	}
-	option := app.ProposalOptions{
-		Option: &app.ProposalOptions_ReleaseEscrowMsg{
-			ReleaseEscrowMsg: &escrow.ReleaseEscrowMsg{
-				Metadata: &weave.Metadata{Schema: 1},
-				EscrowId: *escrowFl,
-				Amount:   amount,
-			},
-		},
+
+	// We must manually assign the message to the right attribute according
+	// to it's type.
+	var option app.ProposalOptions
+	switch msg := msg.(type) {
+	case *cash.SendMsg:
```

Comment:
Hmmm... This is still a manual step, just hidden from the users. Can you add all supported types here now, so we don't forget to update them when we add them as individual commands?

Reply:
I do not know a way to do this. Because the field name differs depending on the type, it must be set dynamically. I think this is not possible.